File: 601-700/0671-Find-Second-Minimum-Value/0671.java
class Solution {
  int ans;
  int rootvalue;

  public int findSecondMinimumValue(TreeNode root) {
    ans = -1;
    rootvalue = root.val;
    dfs(root);
    return ans;
  }

  public void dfs(TreeNode node) {
    if (node == null) {
      return;
    }
    if (ans != -1 && node.val >= ans) {
      return;
    }
    if (node.val > rootvalue) {
      ans = node.val;
    }
    dfs(node.left);
    dfs(node.right);
  }
}

File: 601-700/0671-Find-Second-Minimum-Value/0671.go
func findSecondMinimumValue(root *TreeNode) int {
	ans := -1
	rootVal := root.Val
	var dfs func(*TreeNode)
	dfs = func(node *TreeNode) {
		if node == nil || ans != -1 && node.Val >= ans {
			return
		}
		if node.Val > rootVal {
			ans = node.Val
		}
		dfs(node.Left)
		dfs(node.Right)
	}
	dfs(root)
	return ans
}